cishouseholds/derive.py

```diff
@@ -9,6 +9,28 @@
 from pyspark.sql import Window
 
 
+def assign_random_day_in_month(
+    df: DataFrame, column_name_to_assign: str, month_column: str, year_column: str
+) -> DataFrame:
+    """
+    Assign a random date in a given year and month
+    Parameters
+    ----------
+    df
+    column_name_to_assign
+    month_column
+    year_column
+    """
+    df = df.withColumn("TEMP_DAY", F.lit(1))
+    df = df.withColumn("TEMP_DATE", F.concat_ws("-", year_column, month_column, "TEMP_DAY"))
+    df = df.withColumn("TEMP_DAY", F.round(F.rand() * F.date_format(F.last_day("TEMP_DATE"), "d"), 0).cast("integer"))
```

Comment:
I don't think this will provide a uniform probability of selecting any given day? I think we should [scale the random number according to this](https://stats.stackexchange.com/a/281164) to get a day within the valid range.  I think this would be `(F.rand()/1) * (F.date_format(F.last_day("TEMP_DATE"), "d") - 1) + 1`, but you might want to check this

Reply:
We also need to do some kind of adjustment to use floor rounding, so that all numbers have a equal probability. As I think "half up" rounding used here gives half the chance of producing the min or max day in the range than any other day

Reply:
Current approach may still result in a day value of 0. I've asked methodology to confirm how we should tackle that

Reply:
Agreed approach is to use `F.round()` rounding but where we scale the values between 0.5 and the max day top value (e.g. 31.4999) before rounding.

So I believe this would be `F.round(F.rand() * (F.date_format(F.last_day("TEMP_DATE"), "d") - 0.5001), 0) + 0.5`